e2e/e2e.go

```diff
@@ -287,6 +244,23 @@ func main() { //nolint:funlen,gocyclo // Why: there are no reusable parts to ext
 	// which uses this framework -- but provisions itself.
 	if os.Getenv("SKIP_DEVENV_PROVISION") != "true" {
 		if exec.CommandContext(ctx, "devenv", "--skip-update", "status").Run() != nil {
+			var wg sync.WaitGroup
+			dockerBuilt := false
+			wg.Add(1)
+
+			// Build docker sooner and out of critical path to speed things up.
+			// Docker build in devenv apps deploy . will be superfast then.
+			go func(wg *sync.WaitGroup) {
+				defer wg.Done()
+				log.Info().Msg("Starting early docker build")
+				if err := exec.CommandContext(ctx, "make", "docker-build").Run(); err != nil {
+					log.Warn().Err(err).Msg("Error when running early docker build")
+				} else {
+					log.Info().Msg("Early docker build finished successfully")
+				}
+				dockerBuilt = true
+			}(&wg)
+
 			deps, err := BuildDependenciesList(ctx, conf)
```

Comment:
Now we don't need a dependency list, but just the information if the app is dependent on flagship to deploy the correct snapshot. Potential improvement for the next PR.